### ISSUE TITLE
Says tar is corrupted but it isn't

### ISSUE THREAD

**Issue by binarykitchen** [yarnpkg/yarn#3199]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
`yarn add spider-detector` throws an error saying the remote tar is corrupted or needs to be gunzipped. Here the whole error log:

```
~/c/test ❯❯❯ cat yarn-error.log
Arguments:
  /Users/michael-heuberger/.nvm/versions/node/v4.4.5/bin/node /usr/local/bin/yarn add spider-detector

PATH:
  /Users/michael-heuberger/.nvm/versions/node/v4.4.5/bin:/Users/michael-heuberger/.jenv/shims:/Users/michael-heuberger/.jenv/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:node_modules/.bin

Yarn version:
  0.23.2

Node version:
  4.4.5

Platform:
  darwin x64

npm manifest:
  No manifest

yarn manifest:
  No manifest

Lockfile:
  No lockfile

Trace:
  Error: https://registry.yarnpkg.com/spider-detector/-/spider-detector-1.0.16.tgz: Invalid tar header. Maybe the tar is corrupted or it needs to be gunzipped?
      at Object.exports.decode (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/headers.js:265:40)
      at onheader (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/extract.js:120:39)
      at Extract._write (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/extract.js:243:8)
      at Extract._continue (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/extract.js:207:28)
      at oncontinue (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/extract.js:64:10)
      at ondrain (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/extract.js:84:5)
      at Extract._write (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/extract.js:243:8)
      at Extract._continue (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/extract.js:207:28)
      at oncontinue (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/extract.js:64:10)
      at onunlock (/usr/local/Cellar/yarn/0.23.2/libexec/node_modules/tar-stream/extract.js:70:24)
```

**If the current behavior is a bug, please provide the steps to reproduce.**
Just run that command.

**What is the expected behavior?**
It should install fine.

**Please mention your node.js, yarn and operating system version.**
See above

**Comment by binarykitchen** [comment]
Btw, initially reported here https://github.com/binarykitchen/spider-detector/issues/7

**Comment by Daniel15** [comment]
Looks like this was specific to `spider-detector`, and fixed in 1.0.17.

The error message points to an invalid checksum (see https://github.com/mafintosh/tar-stream/blob/73176274e21b1ee92b06e419099430edb52d0e36/headers.js#L267). This is not a Yarn bug. If you see it again, please file an issue either for the library itself, or for [tar-stream](https://github.com/mafintosh/tar-stream) if the tar file is actually valid.
